### PR TITLE
Update dependencies, switch to ruff format, add CODEOWNERS

### PR DESCRIPTION
**Pull request by willkg**
Minor changes. The more important one here is the `CODEOWNERS` file. The rest is maintenance fluff.

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -50,6 +45,6 @@ commands = pytest {posargs} tests/
 basepython = python3.8
 changedir = {toxinidir}
 commands =
-    black --check setup.py fx_crash_sig tests
-    ruff setup.py fx_crash_sig tests
+    ruff format setup.py fx_crash_sig tests
+    ruff check setup.py fx_crash_sig tests
```
Comment (willkg) [inline pyproject.toml:49]: I like the new `ruff format` thing with 0.1.8. It didn't create any changes here. It's nice. One less dependency plus it re-uses all the ruff options so we don't have to do things like specify line-length and target python twice.

**Review by relud — APPROVED** [review]

**Comment by willkg** [comment]
Thank you!

**Comment by willkg** [comment]
I messed up and missed `Makefile`. I'll do that in a follow-up PR. Bad evil naughty @willkg.

File: pyproject.toml (hunk 1 above)
Comment (biancadanforth) [reply pyproject.toml:49]: Is `ruff` going to replace `black` across all the crash ingestion repos?

**Pull request merged**